### ISSUE TITLE
consider switching to otf

### ISSUE THREAD

**Issue by blobject** [agarick/agave#57]
[alexmyczko said](https://github.com/blobject/agave/issues/55#issue-769591725):
> I've come across these things when looking around on the internet about ttf vs otf.
> https://forum.glyphsapp.com/t/main-difference-between-otf-and-ttf/6743
> http://www.techcybers.com/blog/ttf-otf-or-woff-font-format/#:~:text=OTF%20and%20TTF%20are%20extensions,part%20on%20the%20TrueType%20standard.
> https://www.makeuseof.com/tag/otf-vs-ttf-fonts-one-better/
> https://www.reddit.com/r/typography/comments/ci4nwk/otf_vs_ttf/

**Comment by alexmyczko** [comment]
any thoughts on the links at the very bottom? https://github.com/alexmyczko/fnt (created it because i was annoyed to install fonts by finding url, download manually, copy to the right place)

many years ago there was a page called fonts.debian.net (a font overview), now having that cli tool, i plan to revive something like that but improved with way more fonts...

**Comment by blobject** [comment]
Sorry but I'm not quite sure which links you are referring to. The ones under "Some interesting links" certainly are interesting but I don't see any that pertain to OTF.

With regards to `fnt` itself, well, to be honest, I'm not exactly a fan of the proliferation of "package managers" and I am concerned that the functionality of fnt may overlap with apt, pacman, nixpkgs, xbps, font collection websites, and so on. Especially since font files usually have very loose coupling with the underlying system and should be easy to "port".

Curating less-known fonts can indeed be helpful, but again, one could just package them for apt/pacman/etc., no? If the issue is one of permissions and a difficulty of getting contributions published, perhaps a more effective approach would be to confront the existing packaging system (and/or the attitudes of its maintainers)?

**Comment by alexmyczko** [comment]
> Sorry but I'm not quite sure which links you are referring to. The ones under "Some interesting links" certainly are interesting but I don't see any that pertain to OTF.
> 

Indeed there's nothing that adds on that topic.

> With regards to `fnt` itself, well, to be honest, I'm not exactly a fan of the proliferation of "package managers" and I am concerned that the functionality of fnt may overlap with apt, pacman, nixpkgs, xbps, font collection websites, and so on. Especially since font files usually have very loose coupling with the underlying system and should be easy to "port".
> 

The point is there's so many systems one can use, and packaging a font for a single system doesn't package it
for another. Besides, for example Debian exists since many years (decades of) and only has about 500 font packages. It's immense effort to package huge quantities of fonts for huge quantities of systems, see repology.org. The reason for `fnt` to exist.

> Curating less-known fonts can indeed be helpful, but again, one could just package them for apt/pacman/etc., no? If the issue is one of permissions and a difficulty of getting contributions published, perhaps a more effective approach would be to confront the existing packaging system (and/or the attitudes of its maintainers)?

Basically yes, but if you're in a hurry, `fnt` is a much nicer solution to it, imho. Currently the fonts supported by
fnt are DFSG (Debian) compliant license or Apache/OFL (Googel Web Fonts).

fnt already supports all Linux systems, FreeBSD, macOS, and Haiku OS. It should not be too hard to add more systems to it.

It also allowed me to create a simple web page with font specimen generation, here's a prototype:
https://bootes.ethz.ch/fonts/ for later feeding of into screenshots.debian.net.

**Comment by blobject** [comment]
I see. You make some good points.

> It's immense effort to package huge quantities

The situation reminds me a bit of services like box-look.org that collects and publishes "desktop themes". Or think of all those "wallpaper sites" out there. I guess it's similar with fonts in that no sane linux packaging system would try to package all of those themes and wallpapers individually.

So in this sense, your project could be helpful for people who like to browse and try stuff out.

I guess one way package managers get around the quantity issue (though there don't seem to be many examples outside of artistic things like fonts or themes) is by packaging a group of ~individual items~ finished products into a single package.

> there's nothing that adds on that topic

If it's okay with you, let's try to keep this issue about adopting OTF. I think mentioning your project and discussing it would have been more apt on something like reddit.

**Comment by alexmyczko** [comment]
https://www.typotheque.com/articles/hinting
